runtime/download_test.go

```diff
@@ -0,0 +1,46 @@
+package runtime
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"go4.org/xdgdir"
+)
+
+func Test_downloadFileWithCache(t *testing.T) {
+	contantMock := "Please download me"
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, contantMock)
+	}))
+
+	defer ts.Close()
+	cachePath := filepath.Join(xdgdir.Cache.Path(), "yourbase")
+
+	cacheFilename, err := downloadFileWithCache(context.Background(), ts.URL)
```

Comment:
These files should be removed before the test completes to avoid affecting future runs.